provisioner/terraform/provision.go

```diff
@@ -142,6 +142,11 @@ func (s *server) Plan(
 		return provisionersdk.PlanErrorf("initialize terraform: %s", err)
 	}
 
+	modules, err := getModules(sess.WorkDirectory)
+	if err != nil {
+		return provisionersdk.PlanErrorf("get modules: %s", err)
+	}
```

Comment:
So, if we fail to get the list of modules for the build that we _only need_ for telemetry purposes, we fail to build?

Reply:
Good catch, it'd probably be better to only log an error.

Reply:
On second thought, after the discussion regarding using modules in places other than telemetry, it may make sense to return an error rather than just log. Otherwise any code using module info would have to take into account that it may be missing.

Reply:
> Are there any other possible scenarios I'm missing?

I don't think there are any other scenarios - barring disk read failures and such. In normal operation this should only happen if terraform changes how it stores modules on disk and we upgrade to that version.